planner/physical.go

```diff
@@ -34,17 +38,58 @@ func (pp *physicalPlanner) Plan(spec *PlanSpec) (*PlanSpec, error) {
 		return nil, err
 	}
 
+	// Convert yields into result list
+	// TODO: Implement this via a transformation rule
```

Comment:
Interesting idea to remove `yield` nodes via a transformation rule.  Rules would need to be able to update the `PlanSpec.Results` with the yield name...

Reply:
I've been pondering whether or not plan nodes should contain a pointer to the `PlanSpec` instance that owns them, this makes me think they should.

Reply:
That actually isn't necessary. The planner just has to update the roots (or results) any time a transformation produces a new root node. That logic can be contained in `Plan()`.